File: apiSendCSVFile.js
// Importar bibliotecas necessárias (padrão do Node.js)
const fs = require("fs");
const path = require("path");
const { Blob } = require("buffer");
const mime = require("mime-types");

// Defina o token de autenticação
const token = "TOKEN_API";

// Defina o número de telefone e a mensagem
const body = "Ei Zip-Zop, se liga só nesse arquivo enviado pelo NodeJs!";
const number = "557183196364";

// Caminho do arquivo (em ambiente Node.js)
const filePath = "./files/kapakapa.csv"; // Substitua pelo caminho correto do arquivo no servidor

// URL do endpoint
const url = "URL_API";

// Verifique se o arquivo existe
if (!fs.existsSync(filePath)) {
  console.error("Erro: Arquivo não encontrado:", filePath);
  process.exit(1);
}

// Obtenha o tipo MIME e o nome do arquivo
const mimeType = mime.lookup(filePath) || "application/octet-stream";
const fileName = path.basename(filePath);

// Criar o FormData manualmente com Blobs
const createFormData = () => {
  const boundary = `----WebKitFormBoundary${Math.random().toString(36).substr(2, 16)}`;
  const delimiter = `--${boundary}`;
  const newline = "\r\n";

  let formData = "";

  // Campo 'number'
  formData += `${delimiter}${newline}`;
  formData += `Content-Disposition: form-data; name="number"${newline}${newline}`;
  formData += `${number}${newline}`;

  // Campo 'body'
  formData += `${delimiter}${newline}`;
  formData += `Content-Disposition: form-data; name="body"${newline}${newline}`;
  formData += `${body}${newline}`;

  // Campo 'medias' (arquivo)
  formData += `${delimiter}${newline}`;
  formData += `Content-Disposition: form-data; name="medias"; filename="${fileName}"${newline}`;
  formData += `Content-Type: ${mimeType}${newline}${newline}`;
  const fileBuffer = fs.readFileSync(filePath);
  const fileBlob = new Blob([fileBuffer]);
  formData += fileBlob + newline;

  // Campo 'filename'
  formData += `${delimiter}${newline}`;
  formData += `Content-Disposition: form-data; name="filename"${newline}${newline}`;
  formData += `${fileName}${newline}`;

  // Fechar o formulário
  formData += `${delimiter}--${newline}`;

  return { body: formData, boundary };
};

// Função para enviar a requisição
async function sendFile() {
  try {
    console.log("Iniciando requisição...");

    const { body, boundary } = createFormData();

    const response = await fetch(url, {
      method: "POST",
      headers: {
        Authorization: `Bearer ${token}`,
        "Content-Type": `multipart/form-data; boundary=${boundary}`,
      },
      body,
    });

    if (!response.ok) {
      throw new Error(`Erro na requisição: ${response.status} - ${response.statusText}`);
    }

    // Exibir a resposta da API
    const result = await response.text(); // Alterne para `response.json()` se a API retornar JSON
    console.log("Resposta da API:", response);
  } catch (error) {
    console.error("Erro ao enviar a mensagem:", error);
  } finally {
    console.log("Requisição finalizada.");
  }
}

// Executar a função
sendFile();

File: apiSendCSVFile.py
import os
import mimetypes
import requests

# Mensagem e número do destinatário
body = "Ei Zip-Zop, se liga só nesse arquivo enviado pelo Python!"
number = "557183196364"

# Caminho absoluto do arquivo
dir_path = os.path.dirname(os.path.abspath(__file__))  # Diretório do script atual
file_path = os.path.join(dir_path, "files/kapakapa.csv")  # Caminho completo do arquivo

# URL do endpoint e token
url = 'URL_API'
token = "TOKEN_API"

# Verifique se o arquivo existe
if not os.path.exists(file_path):
    print(f"Erro: Arquivo não encontrado: {file_path}")
    exit(1)

# Obtém o tipo MIME do arquivo
mime_type = mimetypes.guess_type(file_path)[0] or "application/octet-stream"
file_name = os.path.basename(file_path)

# Dados do formulário
with open(file_path, "rb") as file:  # Certifique-se de que o arquivo permanece aberto
    form_data = {
        "number": (None, number),
        "body": (None, body),
        "medias": (file_name, file, mime_type),
        "filename": (None, file_name),
    }

    # Cabeçalhos da requisição
    headers = {
        "Authorization": f"Bearer {token}",
    }

    # Enviar a requisição
    try:
        response = requests.post(url, files=form_data, headers=headers)

        # Verifica se houve erro na requisição
        if response.status_code != 200:
            print(f"Erro na requisição: {response.status_code} - {response.reason}")
            print("Detalhes:", response.text)
            exit(1)

        # Exibe a resposta da API
        print("Resposta da API:")
        print(response.text)

    except requests.RequestException as e:
        print("Erro ao enviar a mensagem:", e)
